community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/ZonedDateTimeLayout.java

```diff
@@ -0,0 +1,114 @@
+
+import org.neo4j.index.internal.gbptree.Layout;
+import org.neo4j.io.pagecache.PageCursor;
+import org.neo4j.kernel.api.schema.index.IndexDescriptor;
+
+/**
+ * {@link Layout} for absolute date times.
+ */
+class ZonedDateTimeLayout extends BaseLayout<ZonedDateTimeSchemaKey>
+{
+    private static final int ZONE_ID_FLAG = 0x0100_0000;
+    private static final int NO_ZONE_ID_FLAG = 0x0000_FFFF;
+
+    public static Layout<ZonedDateTimeSchemaKey,NativeSchemaValue> of( IndexDescriptor descriptor )
+    {
+        return descriptor.type() == IndexDescriptor.Type.UNIQUE ? ZonedDateTimeLayout.UNIQUE : ZonedDateTimeLayout.NON_UNIQUE;
+    }
+
+    private static final ZonedDateTimeLayout UNIQUE = new ZonedDateTimeLayout( "UTdt", 0, 1, ComparableNativeSchemaKey.UNIQUE() );
+    private static final ZonedDateTimeLayout NON_UNIQUE = new ZonedDateTimeLayout( "NTdt", 0, 1, ComparableNativeSchemaKey.NON_UNIQUE() );
+
+    private ZonedDateTimeLayout(
+            String layoutName, int majorVersion, int minorVersion, Comparator<ZonedDateTimeSchemaKey> comparator )
+    {
+        super( layoutName, majorVersion, minorVersion, comparator );
+    }
+
+    @Override
+    public ZonedDateTimeSchemaKey newKey()
+    {
+        return new ZonedDateTimeSchemaKey();
+    }
+
+    @Override
+    public ZonedDateTimeSchemaKey copyKey( ZonedDateTimeSchemaKey key, ZonedDateTimeSchemaKey into )
+    {
+        into.epochSecondUTC = key.epochSecondUTC;
+        into.nanoOfSecond = key.nanoOfSecond;
+        into.zoneId = key.zoneId;
+        into.zoneOffsetSeconds = key.zoneOffsetSeconds;
+        into.setEntityId( key.getEntityId() );
+        into.setCompareId( key.getCompareId() );
+        return into;
+    }
+
+    @Override
+    public int keySize( ZonedDateTimeSchemaKey key )
+    {
+        return ZonedDateTimeSchemaKey.SIZE;
+    }
+
+    @Override
+    public void writeKey( PageCursor cursor, ZonedDateTimeSchemaKey key )
+    {
+        cursor.putLong( key.epochSecondUTC );
+        cursor.putInt( key.nanoOfSecond );
+        if ( key.zoneId >= 0 )
+        {
+            cursor.putInt( key.zoneId | ZONE_ID_FLAG );
+        }
+        else
+        {
+            cursor.putInt( key.zoneOffsetSeconds );
+        }
+        cursor.putLong( key.getEntityId() );
+    }
+
+    @Override
+    public void readKey( PageCursor cursor, ZonedDateTimeSchemaKey into, int keySize )
+    {
+        into.epochSecondUTC = cursor.getLong();
+        into.nanoOfSecond = cursor.getInt();
+        int encodedZone = cursor.getInt();
+        if ( ( encodedZone & ZONE_ID_FLAG ) != 0 )
+        {
+            into.zoneId = (short) ( encodedZone & NO_ZONE_ID_FLAG );
+            into.zoneOffsetSeconds = 0;
+        }
+        else
+        {
+            into.zoneId = -1;
+            into.zoneOffsetSeconds = encodedZone;
+        }
+        into.setEntityId( cursor.getLong() );
+    }
+
+    @Override
+    public boolean fixedSize()
+    {
+        return true;
```

Comment:
Looks like `fixedSize()` method, which is always true could be moved to the BaseLayout too

Reply:
Yupp, did that.